File: cartracker/src/main/webapp/app/model/option/Status.js
Ext.define("CarTracker.model.option.Status",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "longName",
    type : "string"
  }, {
    name : "shortName",
    type : "string"
  }, {
    name : "createDate",
    type : "date",
    dateFormat : "c"
  }, {
    name : "active",
    type : "boolean",
    defaultValue : true
  }, {
    name : "id",
    type : "int",
    useNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "statusOptionService.read",
      create : "statusOptionService.create",
      update : "statusOptionService.update",
      destroy : "statusOptionService.destroy"
    },
    reader : {
      root : "records"
    }
  }
});

File: cartracker/src/main/java/ch/rasc/cartracker/entity/option/Status.java
package ch.rasc.cartracker.entity.option;

import javax.persistence.Entity;

import ch.rasc.extclassgenerator.Model;

@Entity
@Model(value = "CarTracker.model.option.Status", readMethod = "statusOptionService.read",
		createMethod = "statusOptionService.create",
		updateMethod = "statusOptionService.update",
		destroyMethod = "statusOptionService.destroy", paging = true)
public class Status extends BaseOption {
	// nothing special
}
